src/main/java/uk/gov/companieshouse/api/accounts/validation/CreditorsWithinOneYearValidator.java

```diff
@@ -0,0 +1,186 @@
+
+@Component
+public class CreditorsWithinOneYearValidator extends BaseValidator {
+
+    @Value("${invalid.note}")
+    private String invalidNote;
+
+    private static final String CREDITORS_WITHIN_PATH = "$.creditors_within_one_year";
+    private static final String CREDITORS_WITHIN_CURRENT_PERIOD_PATH = CREDITORS_WITHIN_PATH +
+            ".current_period";
+    private static final String CREDITORS_WITHIN_PREVIOUS_PERIOD_PATH = CREDITORS_WITHIN_PATH +
+            ".previous_period";
+    private static final String CREDITORS_WITHIN_CURRENT_PERIOD_TOTAL_PATH =
+            CREDITORS_WITHIN_CURRENT_PERIOD_PATH + ".total";
+    private static final String CREDITORS_WITHIN_PREVIOUS_PERIOD_TOTAL_PATH =
+            CREDITORS_WITHIN_PREVIOUS_PERIOD_PATH + ".total";
+    private static final String CREDITORS_WITHIN_CURRENT_PERIOD_DETAILS_PATH =
+            CREDITORS_WITHIN_CURRENT_PERIOD_PATH + ".details";
+
+    private CompanyService companyService;
+
+    @Autowired
+    public CreditorsWithinOneYearValidator(CompanyService companyService) {
+        this.companyService = companyService;
+    }
+
+    public Errors validateCreditorsWithinOneYear(@Valid CreditorsWithinOneYear creditorsWithinOneYear, Transaction transaction) throws DataException {
+
+        Errors errors = new Errors();
+
+        if (creditorsWithinOneYear != null) {
+
+            if (creditorsWithinOneYear.getCurrentPeriod() != null) {
+
+                validateCurrentPeriod(creditorsWithinOneYear, errors);
+            }
+
+            if (creditorsWithinOneYear.getPreviousPeriod() != null) {
+
+                try {
+
+                    if (companyService.isMultipleYearFiler(transaction)) {
+
+                        validatePreviousPeriod(creditorsWithinOneYear, errors);
+
+                    } else {
+
+                        validateInconsistentFiling(creditorsWithinOneYear, errors);
+                    }
+                } catch (ServiceException e) {
+                    throw new DataException(e.getMessage(), e);
+                }
+            }
+        }
+        return errors;
+    }
+
+    private void validateInconsistentFiling(@Valid CreditorsWithinOneYear creditorsWithinOneYear,
+            Errors errors) {
+        if (creditorsWithinOneYear.getPreviousPeriod().getTotal() != null) {
+
+            addInconsistentDataError(errors,
+                    CREDITORS_WITHIN_PREVIOUS_PERIOD_TOTAL_PATH);
+        }
+    }
+
+    private void validatePreviousPeriod(@Valid CreditorsWithinOneYear creditorsWithinOneYear,
+            Errors errors) {
+        validatePreviousTotalCalculationCorrect(creditorsWithinOneYear, errors);
+        validatePreviousTotalProvidedIfValuesEntered(creditorsWithinOneYear, errors);
+    }
+
+    private void validatePreviousTotalProvidedIfValuesEntered(@Valid CreditorsWithinOneYear creditorsWithinOneYear, Errors errors) {
+        if ((creditorsWithinOneYear.getPreviousPeriod() != null) &&
```

Comment:
Is it possible to reduce the number of creditorsWithinOneYear.getPreviousPeriod() and creditorsWithinOneYear.getCurrentPeriod() by using a local variable, or maybe just passing in the period rather than the full note into the individual validation methods?